src/Processors/Transforms/JoiningTransform.cpp

```diff
@@ -15,7 +15,9 @@ namespace ErrorCodes
 Block JoiningTransform::transformHeader(Block header, const JoinPtr & join)
 {
     ExtraBlockPtr tmp;
+    LOG_DEBUG(&Poco::Logger::get("JoiningTransform"), "Before join block: '{}'", header.dumpStructure());
     join->joinBlock(header, tmp);
+    LOG_DEBUG(&Poco::Logger::get("JoiningTransform"), "After join block: '{}'", header.dumpStructure());
```

Comment:
Do you want to leave these logs?

Reply:
Yes, it's printed once per request and it usually helps to debug how columns and types is changed after join.